### PR TITLE
Check for localStorage availability

### LINKED ISSUE #866: Running Safari with "Block all Cookies" breaks htmx
Running Safari with setting "block all cookies" breaks htmx, due to "insecure operations" Safari is blocking.

Unfortunately since JavaScript and htmx basically runs, the default fallback (e.g. submitting a form the regular way) doesn't get executed instead.

safari

One of them seems to be related to `S(localStorage.getItem("htmx-history-cache"))` – tried both en-/disabling the history using `<meta name="htmx-config" content='{"historyEnabled":false, "refreshOnHistoryMiss": false}'>`, but that didn't change the behavior.

### PR DESCRIPTION
**Pull request by David-Guillot**
Fix #866

### CODE REVIEW

File: src/htmx.js [hunk 1]
```diff
@@ -402,6 +403,17 @@ return (function () {
             }
         }
 
+        function canAccessLocalStorage() {
+            var test = 'test';
```
Comment (1cg) [inline src/htmx.js:407]: Maybe something a little more htmx-specific?  `htmx:localStorageTest` or something like that?  I'd hate to accidentally overwrite something someone really wanted.

**Review by 1cg — CHANGES_REQUESTED** [review]

File: src/htmx.js (hunk 1 above)
Comment (David-Guillot) [reply src/htmx.js:407]: Ah you're right :+1: ! Let me fix that right away.

Comment (David-Guillot) [reply src/htmx.js]: There you go @1cg , I just added a fixup commit. Thanks for your review :slightly_smiling_face:

**Comment by 1cg** [comment]
Ah, can you retarget to `dev` for me?

**Comment by David-Guillot** [comment]
> Ah, can you retarget to `dev` for me?

Done

**Comment by David-Guillot** [comment]
@1cg up on this one, which has been properly rebase against `dev` with the fixup commit fixed-up :wink:

**Comment by David-Guillot** [comment]
@1cg up on this one, which has been properly rebase against `dev` with the fixup commit fixed-up :wink:

**Pull request merged**